system/modules/playos-status.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+let
+  ttyNumber = 8;
+  tty = "tty${toString ttyNumber}";
+  ttyPath = "/dev/${tty}";
+  driverUrl = "http://127.0.0.1:8382";
+
+  script =
+    pkgs.writeShellScriptBin "print-status" ''
+      while :; do
+        screen=$(xrandr --current | grep '*' | awk '{print $1}')
+        network=$(connmanctl technologies | grep 'Type\|Connected')
+        rfid=$(curl -s "${driverUrl}/rfid/readers" | jq -r ".readers" | jq length)
+        controller=$(systemctl is-active playos-controller)
+        printf "\033c"
+        printf "%s\n" \
+          "Screen dimensions: $screen" \
+          "Network connection:" \
+          "$network" \
+          "RFID: $rfid" \
+          "Controller: $controller" \
+          > ${ttyPath}
+        sleep 5
+      done
+    '';
+in
+{
+  config = {
+    console.extraTTYs = [ tty ];
+    systemd.services.playos-status = {
+      environment = {
+        XAUTHORITY = "${config.users.users.play.home}/.Xauthority";
+        DISPLAY = ":0";
+      };
+      path = with pkgs; [
+        connman
+        curl
+        gnugrep
+        gawk
+        jq
+        xorg.xrandr
+      ];
```

Comment:
This is fine, but those could be directly used inside the script, for ex.: `${connman}/bin/connman`.

Reply:
True! I did it like this because think the script is more readable this way.

Reply:
It’s not bad either with `{pkg}/bin/pkg`, because there is syntax highlighting. Now, we would not use it for `printf` for ex. So, this may be more coherent like this.